From 07402d1147229e02010e9c1edceba58255374bee Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:02 -0400
Subject: [ethernet] i40e: Clear recovery pending, if reset failed

Message-id: <1409658042-22689-122-git-send-email-sassmann@redhat.com>
Patchwork-id: 90315
O-Subject: [RHEL7.1 PATCH 121/281] i40e: Clear recovery pending, if reset failed
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>

If pf_reset failed, it becomes necessary to clear recovery pending bit, instead
of falling through the setup process.

Change-ID: Ic1611e6a32741fe3a2782ec7be173cb65e7492ed
Signed-off-by: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit b556540030d4c15fba98219724fb3b5e0bca9c5d)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index f8e1103..618d941 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -5367,8 +5367,10 @@ static void i40e_reset_and_rebuild(struct i40e_pf *pf, bool reinit)
   * because the reset will make them disappear.
   */
  ret = i40e_pf_reset(hw);
- if (ret)
+ if (ret) {
   dev_info(&pf->pdev->dev, "PF reset failed, %d\n", ret);
+  goto end_core_reset;
+ }
  pf->pfr_count++;
 
  if (test_bit(__I40E_DOWN, &pf->state))
-- 
1.7.1